Lesson16/lesson16/gokce/script.js

```diff
@@ -0,0 +1,388 @@
+
+/*
+8. Print Each Character of a String (For-of)
+   - Define a function `printCharacters(str)` that uses a for-of loop
+     to log each character in the string on a separate line.
+*/
+function printCharacters(str) {
+  for (const character of str) {
+    console.log(character);
+  }
+}
+
+
+const exampleString = "Hello World";
+printCharacters(exampleString);
+/*
+9. Sum All Values in an Object
+   - Define a function `sumObjectValues(obj)` that iterates over the
+     properties of an object (using a for-in loop) and sums all numeric values.
+   - Log the sum.
+   - Example: {a: 10, b: 20, c: 5} -> 35
+*/
+
+function sumObjectValues(obj) {
+  let sumKey = 0;
+  for (const key in obj) {
+    sumKey += obj[key];
+  }
+  console.log(sumKey);
+}
+
+
+const person = {
+  name: "Ali",
+  age: 25,
+  height: 175,
+  weight: 70,
+};
+sumObjectValues(person);
+/*
+10. Print Keys of an Object (For-in)
+    - Define a function `printObjectKeys(obj)` that uses a for-in loop
+      to log each key of the object.
+    - Example: { name: "Alice", age: 25 } -> logs "name", then "age"
+*/
+function printObjectKeys(obj) {
+  for (const key in obj) {
+    console.log(key);
+  }
+}
+
+printObjectKeys(person);
+/*
+11. Sum Array Using do-while Loop
+    - Define a function `sumWithDoWhile(numbers)` that uses a do-while loop
+      to sum all numbers in the array.
+    - Log the total.
+*/
+function sumWithDoWhile(numbers) {
+  let total = 0;
+  let index = 0;
+  do {
+    total += numbers([index]);
+    console.log(total);
+  } while (index < numbers.length);
+}
+sumWithDoWhile(numbers);
+sumWithDoWhile([1, 2, 3, 4, 5]);
+
+/*
+12. Remove Duplicates from an Array
+    - Define a function `removeDuplicates(arr)` that loops through the array
+      and creates a new array without duplicate elements.
+    - Hint: you could check if the item is already in the new array before pushing.
+    - Log the new array without duplicates.
+*/
+
+const ex1Array = [1, 2, 2, 3, 4, 4, 5];
+function removeDuplicates(arr) {
+  let removeDup = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (removeDup.indexOf(arr[i] === -1)) {
+      removeDup.push(arr[i]);
+    }
+    console.log(removeDup);
+  }
```

Comment:
Move console log statement outside the for loop so it actually prints the final result
```suggestion
  }
 console.log(removeDup);
```